R/create_table_patient_data_changes_only.R

```diff
@@ -0,0 +1,97 @@
+#' @title Create CSV with longitudinal patient data for a single variable.
+#'
+#' @description
+#' Read in all cleaned patient data CSV and create a single data.frame.
+#' Group this data by id and take only the months when there is a change in the medical data.
+#'
+#'
+#' @param patient_data_files list of CSV files with cleaned patient data from step 2.
+#' @param input_root root directory of the input CSV files.
+#' @param output_root root directory of the output folder.
+#' @param variable name of the column that should be exported.
+#' @param name name used to create the export file name.
+create_table_longitudinal_data <-
+    function(patient_data_files,
+             input_root,
+             output_root,
+             variable,
+             name) {
+        logInfo("Start creating single csv for table create_table_longitudinal_data and variable ", variable)
+
+        dynamic_patient_columns <-
+            c(
+                "blood_pressure_dias_mmhg",
+                "blood_pressure_sys_mmhg",
+                "bmi",
+                "bmi_date",
+                "clinic_code",
+                "country_code",
+                "fbg_updated_date",
+                "fbg_updated_mg",
+                "fbg_updated_mmol",
+                "file_name",
+                "hba1c_updated",
+                "hba1c_updated_exceeds",
+                "hba1c_updated_date",
+                "height",
+                "hospitalisation_cause",
+                "hospitalisation_date",
+                "id",
+                "insulin_regimen",
+                "last_clinic_visit_date",
+                "last_remote_followup_date",
+                "observations",
+                "observations_category",
+                "sheet_name",
+                "status",
+                "support_from_a4d",
+                "testing_frequency",
+                "tracker_month",
+                "tracker_year",
+                "updated_2022_date",
+                "weight"
+            )
+
+        patient_data_list <- list()
+
+        # get the latest static patient data for each tracker file
+        for (patient_file in patient_data_files) {
+            patient_data <- read_csv(
+                file.path(input_root, patient_file),
+                locale = readr::locale(encoding = "UTF-16LE"),
+                show_col_types = F,
+                col_types = "iiinDccDcnnDnncnlnlDncDccDDDccccDccccciDciiiDn",
+                col_select = all_of(dynamic_patient_columns)
```

Comment:
This seems very hard to maintain in case there are ever any changes. I recommend to use a named list with the column type mappings instead.
```R
dynamic_col_types <- list(
     "blood_pressure_dias_mmhg" = "i",
      "blood_pressure_sys_mmhg"= "i",
      ...
)
```
That way one could use the following expressions for `col_types` and `col_select`.
```R
col_types = do.call(paste0,dynamic_col_types)
col_select = all_of(names(dynamic_col_types))
```

Reply:
Wow cool that helps really I didn't have much time to think about a better solution but this makes much more sense and improves maintainability a lot so thanks again

Reply:
I just remembered that this will be completely obsolete once we have the parquet files!